test/language/expressions/object/computed-property-evaluation-order.js

```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2016 Michael Ficarra. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-object-initializer-runtime-semantics-propertydefinitionevaluation
+description: >
+    Evaluation of PropertyDefinitionList occurs in order, and each
+    PropertyDefinition's PropertyName is evaluated before its
+    AssignmentExpression.
+---*/
+
+var counter = 0;
+var o = {
+  [++counter]: ++counter,
+  [++counter]: ++counter,
+  [++counter]: ++counter,
+}
+
+var keys = Object.keys(o).sort();
```

Comment:
We have to assume Object.keys is acting according to the specs, which the order is defined.

If this built-in becomes a problem, I believe pure assertions over the expected properties will solve this in a better fashion:

```js
assert.sameValue(o[1], 2);
assert.sameValue(o[3], 4);
assert.sameValue(o[5], 6);
```

If necessary, Object.keys could still be used to compare the assigned keys, which I believe is overkill, but still a valid test:

```js
assert(compareArray(Object.keys(o), [1, 3, 5]));
```